workflow/schemas.py

```diff
@@ -0,0 +1,616 @@
+        ),
+        Literal("kappa", description="Unknown!"): float,
+        Literal("qfexp", description="Q frequency exponent"): float,
+        Literal("rvfac", description="Rupture velocity factor (rupture : Vs)"): And(
+            float, is_non_negative
+        ),
+        Literal("rvfac_shal", description="rvfac shallow fault multiplier"): float,
+        Literal("rvfac_deep", description="rvfac deep fault multiplier"): float,
+        Literal("czero", description="C0 coefficient"): float,
+        Literal("calpha", description="Ca coefficient"): float,
+        Literal("mom", description="Seimic moment (or null, to infer value)"): Or(
+            float, None
+        ),
+        Literal("rupv", description="Rupture velocity (or binary default)"): Or(
+            float, None
+        ),
+        Literal("site_specific", description="Enable site-specific calculation"): bool,
+        Literal("vs_moho", description="vs of moho layer"): float,
+        Literal("fa_sig1", "Fourier amplitude uncertainty (1)"): float,
+        Literal("fa_sig2", description="Fourier amplitude uncertainty (2)"): float,
+        Literal("rv_sig1", description="Rupture velocity uncertainty"): And(
+            float, is_non_negative
+        ),
+        Literal(
+            "path_dur",
+            description="path duration model. 0: GP2010, 1: WUS modification trail/errol, 2: ENA modificiation trial/error"
+            ", 11: WUS formutian of BT2014, 12: ENA formulation of BT2015. Models 11 and 12 overpredict for multiple rays.",
+        ): Or(0, 1, 2, 11, 12),
+        Literal("dpath_pert", description="Log of path duration multiplier"): float,
+        Literal(
+            "stress_parameter_adjustment_tect_type",
+            description="Adjustment option 0 = off, 1 = active tectonic, 2 = stable continent",
+        ): Or(0, 1, 2),
+        Literal(
+            "stress_parameter_adjustment_target_magnitude",
+            description="Target magnitude (or inferred if null)",
+        ): Or(float, None),
+        Literal(
+            "stress_parameter_adjustment_fault_area", "Fault area (or inferred if null)"
+        ): Or(float, None),
+        Literal("seed", description="HF seed"): int,
+        Literal("stoch_dx", description="Stoch file dx"): And(float, is_positive),
+        Literal("stoch_dy", description="Stoch file dy"): And(float, is_positive),
+    }
+)
+
+EMOD3D_PARAMETERS_SCHEMA = Schema(
+    {
+        "all_in_one": int,
+        "bfilt": int,
+        "bforce": int,
+        "dampwidth": int,
+        "dblcpl": int,
+        "dmodfile": str,
+        "dtts": int,
+        "dump_itinc": int,
+        "dxout": int,
+        "dxts": int,
+        "dyout": int,
+        "dyts": int,
+        "dzout": int,
+        "dzts": int,
+        "elas_only": int,
+        "enable_output_dump": int,
+        "enable_restart": int,
+        "ffault": int,
+        "fhi": float,
+        "fmax": float,
+        "fmin": float,
```

Comment:
Surely some of these can be test for is_positive or other tests etc. Even though we don't know exactly what all of these parameters do. These 2 as an example

Reply:
Since the merge of #5 I have a better idea of what these values do so I will go through and give them sensible bounds.